backend/src/v2/driver/driver.go

```diff
@@ -1143,19 +1203,56 @@ func resolveInputs(ctx context.Context, dag *metadata.DAG, iterationIndex *int,
 			if err != nil {
 				return nil, paramError(err)
 			}
+
+			// The producer is the task that produces the output that we need to
+			// consume.
 			producer, ok := tasks[taskOutput.GetProducerTask()]
-			if !ok {
-				return nil, paramError(fmt.Errorf("cannot find producer task %q", taskOutput.GetProducerTask()))
-			}
-			_, outputs, err := producer.GetParameters()
+
+			glog.V(4).Info("producer: ", producer)
+
+			// Get the producer's outputs.
+			_, producerOutputs, err := producer.GetParameters()
```

Comment:
This whole producer subtask lookup should be decomposed and called recursively in case the subtask is *also* a DAG!

Reply:
Just pushed up a commit that implements this and confirmed that it works with subdags of subdags. Here's is the example code to validate it:

```python
from kfp import dsl
from kfp.client import Client

@dsl.component
def small_comp() -> str:
    return "privet"

@dsl.component
def large_comp(input: str):
    print("input :", input)


@dsl.pipeline
def small_matroushka_doll() -> str:
    task = small_comp()
    task.set_caching_options(False)
    return task.output

@dsl.pipeline
def medium_matroushka_doll() -> str:
    dag_task = small_matroushka_doll()
    dag_task.set_caching_options(False)
    return dag_task.output

@dsl.pipeline
def large_matroushka_doll():
    dag_task = medium_matroushka_doll()
    task = large_comp(input=dag_task.output)
    task.set_caching_options(False)
    dag_task.set_caching_options(False)


if __name__ == "__main__":
    client = Client()

    run = client.create_run_from_pipeline_func(
        pipeline_func=large_matroushka_doll,
        enable_caching=False,
    )
```

PS. I hate matroushka dolls, they're so full of themselves.